### PR TITLE
refactor: migrate test-infra to testify for mainly window_func_test.go

### LINKED ISSUE #28502: migrate test-infra to testify for `window_func_test.go`

### LINKED ISSUE #28503: migrate test-infra to testify for `row_number_test.go`

### LINKED ISSUE #28507: migrate test-infra to testify for `func_cume_dist_test.go`

### LINKED ISSUE #28512: migrate test-infra to testify for `func_lead_lag_test.go`

### LINKED ISSUE #28514: migrate test-infra to testify for `func_ntile_test.go`

### LINKED ISSUE #28515: migrate test-infra to testify for `func_percent_rank_test.go`

### LINKED ISSUE #28517: migrate test-infra to testify for `func_rank_test.go`

### LINKED ISSUE #28521: migrate test-infra to testify for `func_value_test.go`

### PR DESCRIPTION
**Pull request by DiFronzo**
### What problem does this PR solve?

* close #28502
* close #28507
* close #28512 
* close #28514 
* close #28515 
* close #28517
* close #28521 
* close #28503 

Problem Summary:
Migrate test-infra to testify for `window_func_test.go`. BUT in the file `window_func_test.go` there are the functions `testWindowAggMemFunc()` and `testWindowFunc()` that has been changed to use testify. This brakes the code in the other tests that calls one of the previously mentioned functions. So, I had to make some changes to some other files too, listed below.

Other files that are also included and finished:
 - row_number_test.go
 - func_value_test.go
 - func_rank_test.go
 - func_percent_rank_test.go
 - func_ntile_test.go
 - func_lead_lag_test.go
 - func_cume_dist_test.go

### What is changed and how it works?

What's Changed:
Migrate test-infra to testify for mainly `window_func_test.go`.

How it Works:
All tests passed.

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by tisonkun** [comment]
> Other files that are also included and finished

@DiFronzo thanks for your contribution! Can you also assign those issues to you, and close them in this PR by:

* close #xxx
* close #yyy
* ...

?

**Comment by DiFronzo** [comment]
Issue Numbers:
* close #28507
* close #28512
* close #28514 
* close #28515 
* close #28517

File: executor/aggfuncs/window_func_test.go [hunk 1]
```diff
@@ -54,52 +57,54 @@ type windowMemTest struct {
...
 
 	i := 0
 	iter := chunk.NewIterator4Chunk(srcChk)
 	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
-		memDelta, err = finalFunc.UpdatePartialResult(s.ctx, []chunk.Row{row}, finalPr)
-		c.Assert(err, IsNil)
-		c.Assert(memDelta, Equals, updateMemDeltas[i])
+		memDelta, err = finalFunc.UpdatePartialResult(ctx, []chunk.Row{row}, finalPr)
+		require.NoError(t, err)
+		require.Equal(t, memDelta, updateMemDeltas[i])
```
Comment (tisonkun) [inline executor/aggfuncs/window_func_test.go:107]:
```suggestion
		require.Equal(t, updateMemDeltas[i], memDelta)
```

**Review by tisonkun — COMMENTED** [review]
Comments inline. Also update the description to leverage GitHub magic for closing issues.

**Review by pingyu — COMMENTED** [review]
The same suggestion as @tisonkun , others LGTM.

Nit: suggest add [`t.Parallel()`](https://pkg.go.dev/testing#T.Parallel) as first line of unit test function, to get a better execution efficiency.

File: executor/aggfuncs/func_cume_dist_test.go [hunk 2]
```diff
@@ -15,13 +15,14 @@
...
-	. "github.com/pingcap/check"
+	"testing"
+
 	"github.com/pingcap/parser/ast"
 	"github.com/pingcap/parser/mysql"
 	"github.com/pingcap/tidb/executor/aggfuncs"
 )
 
-func (s *testSuite) TestMemCumeDist(c *C) {
+func TestMemCumeDist(t *testing.T) {
```
Comment (pingyu) [inline executor/aggfuncs/func_cume_dist_test.go:25]:
```suggestion
func TestMemCumeDist(t *testing.T) {
	t.Parallel()
	
```

File: executor/aggfuncs/window_func_test.go [hunk 3]
```diff
@@ -54,52 +57,54 @@ type windowMemTest struct {
...
 	iter := chunk.NewIterator4Chunk(srcChk)
 	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
-		_, err = finalFunc.UpdatePartialResult(s.ctx, []chunk.Row{row}, finalPr)
-		c.Assert(err, IsNil)
+		_, err = finalFunc.UpdatePartialResult(ctx, []chunk.Row{row}, finalPr)
+		require.NoError(t, err)
 	}
 
-	c.Assert(p.numRows, Equals, len(p.results))
+	require.Equal(t, len(p.results), p.numRows)
```
Comment (mmyj) [inline executor/aggfuncs/window_func_test.go:76]: `require.Len(t, p.results, p.numRows)`

**Comment by mmyj** [comment]
It seems that all unit test can be paralleled.

**Review by mmyj — APPROVED** [review]
lgtm

**Comment by mmyj** [comment]
PTAL @tisonkun, we need another one approving.

**Review by tisonkun — APPROVED** [review]
LGTM.

Thanks for your contribution @DiFronzo !

**Pull request merged**